### ISSUE TITLE
Implement noise-free training

### ISSUE THREAD

**Issue by GageDeZoort** [gnn-tracking/gnn_tracking#186]
It occurs to me that the noise points might hinder training. We should try to train without them, then test the robustness of the learned model to varying amounts of noise.

**Comment by klieret** [comment]
So we would mask them for the loss functions? That could be a very simple option to the loss function classes. We still want to keep the noise in the tests steps with the clustering to have realistic FOMs.

**Comment by GageDeZoort** [comment]
Exactly, yes, we would ask for some % of the noise hits to remain in each test step.

**Comment by klieret** [comment]
Ah, evaluating the test metrics with different percentages of noise...

**Comment by klieret** [comment]
Excluding noise from the losses seems very similar to the pt threshold for loss functions to me... And there it doesn't seem obvious to me a priori that it helps. But we can certainly try.

> Ah, evaluating the test metrics with different percentages of noise...
This would mean spending more time not training. Not sure if that is worth it. Because if we see that it cannot handle the full noise, what value would the model be anyway?

**Comment by klieret** [comment]
This is how the loss pt threshold was implemented for reference: #123

**Comment by klieret** [comment]
Ah, so the attractive/repulsive loss is already without noise! It's just the background (?) and EC losses that currently take noise
